.github/workflows/docker_simulate-sce.yml

```diff
@@ -23,59 +19,40 @@ on:
       - analyses/simulate-sce/.dockerignore
       - analyses/simulate-sce/renv.lock
       - analyses/simulate-sce/conda-lock.yml
-
-permissions:
-  id-token: write # This is required for requesting the JWT
-  contents: read # This is required for actions/checkout
+  workflow_dispatch:
+    inputs:
+      push-ecr:
+        description: "Push to AWS ECR"
+        type: boolean
+        required: false
+  workflow_call:
+    inputs:
+      push-ecr:
+        description: "Push to AWS ECR"
+        type: boolean
 
 jobs:
-  build:
-    name: Build Docker Image
+  test-build:
+    name: Test Build Docker Image
+    if: github.event_name == 'pull_request' || (contains(github.event_name, 'workflow_') && !inputs.push-ecr)
     runs-on: ubuntu-latest
 
     steps:
-      - name: Configure AWS Credentials
-        uses: aws-actions/configure-aws-credentials@v4
-        with:
-          role-to-assume: arn:aws:iam::992382809252:role/GithubOpenId
-          role-session-name: githubActionSession
-          aws-region: us-east-1
-
-      - name: Log in to Amazon ECR
-        id: login-ecr
-        if: github.event_name == 'push'
-        uses: aws-actions/amazon-ecr-login@v2
-        with:
-          registry-type: public
-
-      - name: Create ECR repository if needed
-        id: create-ecr
-        if: github.event_name == 'push'
-        run: |
-          aws ecr-public describe-repositories --repository-names ${{ env.MODULE }} \
-          || aws ecr-public create-repository --repository-name ${{ env.MODULE }}
-
       - name: Set up Docker Buildx
         uses: docker/setup-buildx-action@v3
 
-      - name: Docker metadata
-        id: meta
-        uses: docker/metadata-action@v5
-        with:
-          images: public.ecr.aws/${{ env.REGISTRY_ALIAS }}/${{ env.MODULE }}
-          # tag with 'latest' for main branch pushes, semantic version for releases/tags, and the branch name for PRs
-          # PR tags are just for convenience and to avoid warnings, as those builds are not pushed
-          tags: |
-            type=raw,value=latest,enable={{is_default_branch}}
-            type=semver,pattern={{raw}}
-            type=ref,event=pr
-
-      - name: Build and push image
+      - name: Build image
         uses: docker/build-push-action@v5
         with:
-          context: "{{defaultContext}}:${{ env.MODULE_PATH }}"
-          push: ${{ github.event_name == 'push'}}
-          tags: ${{ steps.meta.outputs.tags }}
-          labels: ${{ steps.meta.outputs.labels }}
+          context: "{{defaultContext}}:analyses/simulate-sce"
+          push: false
           cache-from: type=gha
           cache-to: type=gha,mode=max
+
+  build-push:
+    name: Build and Push Docker Image
+    uses: ./.github/workflows/build-push-docker-module.yml
+    if: github.event_name == 'push' || inputs.push-ecr
+    with:
+      module: simulate-sce
```

Comment:
This field would get populated at template time, yea?

Reply:
Looks like yes from what you just pushed!

Reply:
yup!